pkg/api/types.go

```diff
@@ -421,6 +425,19 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
```

Comment:
So hypothetically, an admin might run gluster in namespace "foo" and have a real service "gluster" (headless or no).  They then want to use volumes from that gluster service in other namespaces.  So an admin would automate / manually create persistent volumes that point to that gluster cluster.  The volume settings would be "use the gluster cluster in namespace foo with name gluster".  When a volume source is created for that persistent volume, it would be referencing that service.

----- Original Message -----
> > @@ -421,6 +425,19 @@ type NFSVolumeSource struct {
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> >  }
> >  
> > +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the
> > lifetime of a pod
> > +type GlusterfsVolumeSource struct {
> > +	// Required: EndpointsName is the endpoint name that details Glusterfs
> > topology
> > +	EndpointsName string `json:"endpoints"`
> 
> We do not currently have any concept of non-namespaced endpoints or services.
> Are we going to accumulate these things in a random namespace and then
> violate the cross-namespace principles?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6174/files#r28000861